bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/widgets/FileDialogResult.java

```diff
@@ -0,0 +1,62 @@
+package org.eclipse.swt.widgets;
+
+/**
+ * A representation of the result of file dialogs such as the {@link FileDialog}
+ * or {@link DirectoryDialog}.
+ *
+ * @since 3.125
+ */
+public class FileDialogResult {
+	public static enum State {
+		/**
+		 * The user has successfully selected a file or folder
+		 */
+		SELECTED,
+		/**
+		 * The user has canceled the dialog
+		 */
+		CANCELED,
+		/**
+		 * An unspecified error occurred while the dialog was open
+		 */
+		UNKNOWN_ERROR
+	}
+
+	private String path;
+
+	private State state;
+
+	private FileDialogResult(String path, State state) {
+		this.path = path;
+		this.state = state;
+	}
+
+	/**
+	 * {@return a string describing the absolute path of the selected file or folder
+	 * or <code>null</code> if the dialog was canceled or if there was an error}
+	 */
+	public String getSelectedPath() {
```

Comment:
```suggestion
	public Optional<String> getPath() throws SWTException {
```

this should throw exception when passed in an error state and an empty optional if the operation was canceled, that way one can simply always call this method and don't need to fiddle around will `null` checks.

also "Selected" is superfluous here as there is no other path anyways.

Reply:
I like the idea of returning an Optional. I have implemented that together with the simplified name in bff9b11.
I don't think we should thrown an `SWTException` here, as the errors that may occur are no SWT-internal fails, but something related to semantics of the dialog, so that returning a proper path was not possible (like a too long pathname on Windows). So they are still kind of "valid results", at least in terms of the contract between caller (opening the dialog) and callee (the dialog), which is why I would not express them in terms of exceptions at all.